### DISCUSSION TITLE
Integrate Google Analytics Reporting API from plugin

### DISCUSSION THREAD

**Discussion by dinigo** [apache/airflow#30790] [Ideas]
Short: Integrate airflow [plugin](https://github.com/airflow-plugins/google_analytics_plugin) , notified in this [issue](https://github.com/airflow-plugins/google_analytics_plugin/issues/18) on their issue tracker.

Before the official (beta for now) operators for Google Analytics landed in the official airflow repo a number of custom plugins appeared in the community. This repo gathers a couple of them, quite spread.

Who's responsibility should it be to merge this?
Can we simply copy-paste the code? I think it would be better to keep the commit-tree but not sure how it applies to airflow
Is Google working in this issue on it's own or does this belong to the airflow community?

**Comment by turbaszek** [comment]
> Is Google working in this issue on it's own or does this belong to the airflow community?

The repository you linked doesn't seem to belong to Google. If there's something maintained by Google then it's this repo: https://github.com/google/orchestra


Does already existing Analytics operators solved the problem?
https://github.com/apache/airflow/blob/master/airflow/providers/google/marketing_platform/operators/analytics.py

**Comment by dinigo** [comment]
No, we lack the reporting API that lets you compose a report and download a dataset for this report.

It's the easiest way to take Data out of Google Analytics. it was there long before  the BigQiery export so a lot of business logic for a lot of companies relay in this export. I've seen it "Cron"ified very frequently.

**Comment by dinigo** [comment]
I just checked the Orchestra project, which seems to be a "plug'n play" version of the Google operators we have in the master (I suppose it will be obsolete once the operators backport is officially released from apache-airflow). As it is basically a copy of the code in airflow/providers/google it also lacks this API.

**Comment by turbaszek** [comment]
It's true that we get some help from Orchestra projects. Airflow is open source so I encourage you to open a PR with operators / hooks you require. I am happy to help with review / testing :)
